Polyhedron/demo/Polyhedron/triangulate_primitive.h

```diff
@@ -141,8 +141,8 @@ class FacetTriangulator
     }
     if(last_inserted == typename CDT::Vertex_handle())
       return false;
-    
-    cdt->insert_constraint(previous, first);
+    if(previous != first)
+      cdt->insert_constraint(previous, first);
```

Comment:
I am really surprised, that the code of the CDT does not handle that case by its own. Are you sure that this test fixes a bug?

Reply:
Well, the code of the CDT only uses an assertion. Before this test it crashed, after it doesn't, so I'm pretty sure, yes.

Reply:
Then that might be a bug of the code of CDT, because [the documentation does not document that precondition][1].

[1]: https://doc.cgal.org/4.13/Triangulation_2/classCGAL_1_1Constrained__Delaunay__triangulation__2.html#acd1f7edafecd48ea978e43c98c87172e

Reply:
Right here: https://github.com/CGAL/cgal/blob/0d4b7185bd7d73d21e96ea8f706c9a9d2ed0d269/Triangulation_2/include/CGAL/Constrained_triangulation_2.h#L296